pkg/forklift-api/webhooks/validating-webhook/admitters/secret-admitter.go

```diff
@@ -71,12 +68,36 @@ func (admitter *SecretAdmitter) Admit(ar *admissionv1.AdmissionReview) *admissio
 			if err != nil {
 				log.Info("Connection test failed, yet passing", "status", status, "error", err.Error())
 			} else {
-				log.Info("Test succeeded, passing")
-			}
-			return &admissionv1.AdmissionResponse{
-				Allowed: true,
+				// Perform validation for the provided engine CA certificate,
+				// initiate dummy image transfer to make sure the certificate is correct and that the migration will work in a later stage.
+				if createdForProviderType == "ovirt" {
+					if isCaOK, err := ovirt.TestDownloadOvfStore(secret, log); isCaOK {
+						if err != nil {
+							log.Info("CA certificate test failed, yet passing", "error", err.Error())
+						} else {
+							log.Info("Test credentials and CA certificate succeeded, passing")
+						}
+					} else {
+						log.Info("Test credentials succeeded but engine CA certificate is not valid, passing")
+						return &admissionv1.AdmissionResponse{
+							Allowed: true,
+							Result: &metav1.Status{
+								Code:    http.StatusForbidden,
+								Message: "Test credentials succeeded but engine CA certificate is not valid",
+							},
+						}
+					}
+				} else {
+					log.Info("Test credentials succeeded, passing")
```

Comment:
well, that's a bit misleading for non-ovirt providers since we don't test their credentials (yet?), no?

Reply:
this is part of the original admitter (not related to my changes), because we only check the label `createdForProviderType` exists and not the provider type itself for all of the providers, and the matching Test() function should be called (IIUC), if it's not yet supported then we should switch the check in the beginning to be only for ovirt.

Reply:
ah you're right